toothpick-runtime/src/main/java/toothpick/Configuration.java

```diff
@@ -4,100 +4,182 @@
+      return MemberInjectorRegistryLocator.getMemberInjectorUsingReflection(clazz);
+    }
+  }
 
+  private static class DevelopmentConfiguration extends Configuration {
+    // We need a LIFO structure here, but stack is thread safe and we use thread local,
+    // so this property is overkill and LinkedHashSet is faster on retrieval.
+    private ThreadLocal<LinkedHashSet<Class>> cycleDetectionStack = new ThreadLocal<LinkedHashSet<Class>>() {
       @Override
-      void checkCyclesEnd(Class clazz) {
-        cycleDetectionStack.get().remove(clazz);
+      protected LinkedHashSet<Class> initialValue() {
+        return new LinkedHashSet<>();
       }
     };
-  }
 
-  public static Configuration production() {
-    return new Configuration() {
-      @Override
-      void checkIllegalBinding(Binding binding) {
-        //do nothing
+    @Override
+    public void checkIllegalBinding(Binding binding) {
+      Class<?> clazz;
+      switch (binding.getMode()) {
+        case SIMPLE:
+          clazz = binding.getKey();
+          break;
+        case CLASS:
+          clazz = binding.getImplementationClass();
+          break;
+        case PROVIDER_CLASS:
+          clazz = binding.getProviderClass();
+          break;
+        default:
+          return;
       }
 
-      @Override
-      void checkCyclesStart(Class clazz) {
-        //do nothing
+      for (Annotation annotation : clazz.getAnnotations()) {
+        if (annotation.annotationType().isAnnotationPresent(javax.inject.Scope.class)) {
+          throw new IllegalBindingException(format("Class %s cannot be bound. It has an scope annotation", clazz.getName()));
+        }
       }
+    }
 
-      @Override
-      void checkCyclesEnd(Class clazz) {
-        //do nothing
+    @Override
+    public void checkCyclesStart(Class clazz) {
+      if (cycleDetectionStack.get().contains(clazz)) {
+        throw new CyclicDependencyException(new ArrayList<>(cycleDetectionStack.get()), clazz);
       }
-    };
+
+      cycleDetectionStack.get().add(clazz);
+    }
+
+    @Override
+    public void checkCyclesEnd(Class clazz) {
+      cycleDetectionStack.get().remove(clazz);
+    }
+
+    @Override
+    public <T> Factory<T> getFactory(Class<T> clazz) {
+      return FactoryRegistryLocator.getFactoryUsingReflection(clazz);
+    }
+
+    @Override
+    public <T> MemberInjector<T> getMemberInjector(Class<T> clazz) {
+      return MemberInjectorRegistryLocator.getMemberInjectorUsingReflection(clazz);
+    }
   }
 
-  /**
-   * Allows to pass custom configurations.
-   *
-   * @param configuration the configuration to use
-   */
-  static void setConfiguration(Configuration configuration) {
-    instance = configuration;
+  private static class ProductionConfiguration extends BaseConfiguration {
+    @Override
+    public <T> Factory<T> getFactory(Class<T> clazz) {
+      return FactoryRegistryLocator.getFactoryUsingReflection(clazz);
+    }
```

Comment:
I don't understand your question. Production is the default configuration and the PR is about not using registries by default.